client/my-sites/posts/post-share.jsx

```diff
@@ -212,6 +226,7 @@ export default connect(
 		const userId = getCurrentUserId( state );
 
 		return {
+			planHasRepublicizeFeature: hasFeature( state, siteId, FEATURE_REPUBLICIZE ),
```

Comment:
Since we'll be continuing work on this, I'm OK with this check since we'll likely update this in the future. However, let's warn other devs of the pitfalls, and update the jsdoc that this cannot be trusted as a feature check for the reasons above:

We have usages here in state/sites/selectors:
https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/plans/selectors.js#L217-L229

Also in lib/plans:
https://github.com/Automattic/wp-calypso/blob/master/client/lib/plans/index.js#L87-L91